### PR TITLE
Add option to allow all keybindings in the navigator

### PR DESCRIPTION
**Pull request by Lythenas**
This allows executing all actions while the "navigator" (minimap shown while switching windows) is open. If the action would not be allowed before, it will close the navigator and then execute the action.

This works only for the paperwm minimap that is shown when pressing e.g. `<Super>+Left`. Not for the minimap that is shown by gnome-shell (e.g. `<Alt>+<Tab>`).

Also unfortunately in my testing it only works for keybinding that have been registered with paperwm and not for arbitrary gnome-shell keybindings (see also #268 for this).

Also after doing e.g. `<Super>+Left` and then switching monitors with `<Super>+<Shift>+Left` I get the following warning in the logs:

```
Window manager warning: META_CURRENT_TIME used to choose focus window; focus window may not be correct.
```

But apart from the warning everything works as expected. I also found this issue about the warning: https://gitlab.gnome.org/GNOME/mutter/-/issues/1066

Example usage that was annoying before: Navigate to a window and move it to the other monitor. I.e. `<Super>+Left` followed by `<Super>+<Ctrl>+Left`. This now works without letting go of `<Super>`.

### CODE REVIEW

**Comment by jtaala** [comment]
Cool! - any tips for how you use this? (e.g. what actions do you like doing while minimap is open just so I can test them).

File: schemas/org.gnome.shell.extensions.paperwm.gschema.xml
Comment (jtaala) [inline schemas/org.gnome.shell.extensions.paperwm.gschema.xml]: Do we need this to be an user settable option (with a default of false)?

Is there a reason why we wouldn't want this behaviour by default?  e.g. I can't think of a reason why would not want other shortcuts executable while minimap is open.

**Review by jtaala — COMMENTED** [review]
Looks good - see comment about just having this as default behaviour.

Re the warning - it's a warning and doesn't appear to have any other negative impacts and action appears to execute correctly :man_shrugging:

**Comment by jtaala** [comment]
> Cool! - any tips for how you use this? (e.g. what actions do you like doing while minimap is open just so I can test them).

Never mind - saw an example in the code (switching monitors works fine).

File: schemas/org.gnome.shell.extensions.paperwm.gschema.xml
Comment (Lythenas) [reply schemas/org.gnome.shell.extensions.paperwm.gschema.xml]: I don't know the original reason for separating the keybindings but IMO we can also just change the behavior and not create a user setting for this.

The only breakage I can think of (not tested) is that I think it would be possible (in user.js) to create a keybinding that is only active in the navigator but has the same keys as another keybinding. But I think you can work around that because the handler get passed `navigator` and I think you also globally check if the navigator is open in your custom handler.

Comment (jtaala) [reply schemas/org.gnome.shell.extensions.paperwm.gschema.xml]: Agree, would make this default behaviour.  Test out your hypothesis as much as you can - but with `user.js`, users can implement unintended things/behaviours (including code that would completely break PaperWM) so  I wouldn't let concerns about users potentially implementing something in there stop us from simplifying, testing, and improving PaperWM with this PR.

Make changes you see fit (e.g. making this default) and I can help with testing this.

Comment (Lythenas) [reply schemas/org.gnome.shell.extensions.paperwm.gschema.xml]: I removed the option and just made this the new behavior.

**Comment by Lythenas** [comment]
Unfortunately I haven't found a way to activate arbitrary gnome actions. In my testing only other PaperWM keybindings worked. If you have any ideas here I can investigate further. Otherwise I think this is already a good improvement on the current behavior.

**Review by jtaala — APPROVED** [review]
Approved.

**Comment by jtaala** [comment]
One thing I noticed is that cycling through window widths while minmap is showing  (I have incrementing window widths set to `ALT+E`)  causes the minimap to show some weird empty space where the window portion was (before resize) - that's not an issue with this functionality though (the fact that we can now do this while navigator is showing proves this works).

I might try to fix navigator a bit here.

**Comment by jtaala** [comment]
> I might try to fix navigator a bit here.

Hey @Lythenas, so I just pushed a fix to the issue I mentioned.  Basically it just resets/refreshes the minimap when windows resize (if minimap is showing that is).  Just means if minimap will be shown correctly after a window resize event

I'm ready to merge this now once you approve.

Cheers.

**Comment by Lythenas** [comment]
Good catch. I haven't tried that particular shortcut. Your fix works perfectly. Feel free to merge, unfortunately I still can't merge.

**Comment by jtaala** [comment]
> unfortunately I still can't merge.

You should be able to merge now.  Give it a go when ready.  Let me know if still have issues.

**Comment by jtaala** [comment]
> > unfortunately I still can't merge.
> 
> You should be able to merge now. Give it a go when ready. Let me know if still have issues.

After you approve my last push that is.

**Comment by Lythenas** [comment]
> After you approve my last push that is.

Ah, I see. Because you made the last commit, you also can't approve. Is that correct?

Unfortunately Github does not allow me to approve my own PR... So @smichel17 pls help^^

**Comment by jtaala** [comment]
> Unfortunately Github does not allow me to approve my own PR... So @smichel17 pls help^^

All good, I can push it.

**Pull request merged**
